include/psen_scan_v2/scanner_state_machine_def.h

```diff
@@ -168,15 +178,42 @@ inline void ScannerProtocolDef::handleMonitoringFrameTimeout(const scanner_event
 
 //+++++++++++++++++++++++++++++++++ Guards ++++++++++++++++++++++++++++++++++++
 
+inline void ScannerProtocolDef::checkForInternalErrors(const scanner_reply::Message& msg)
+{
+  // LCOV_EXCL_START
+  if (msg.result() != scanner_reply::Message::OperationResult::accepted)
+  {
+    PSENSCAN_ERROR("StateMachine", "Received reply with non-successful result code.");
+    if (msg.result() == scanner_reply::Message::OperationResult::refused)
+    {
+      if (msg.type() == scanner_reply::Message::Type::stop)
+      {
+        PSENSCAN_ERROR("StateMachine", "Stop request refused by device.");
+      }
+      if (msg.type() == scanner_reply::Message::Type::start)
+      {
+        PSENSCAN_ERROR("StateMachine", "Start request refused by device.");
+      }
+    }
+    else
+    {
+      PSENSCAN_ERROR("StateMachine", "Unknown operation result code.");
+    }
+  }
+  // LCOV_EXCL_STOP
+}
+
 inline bool ScannerProtocolDef::isStartReply(scanner_events::RawReplyReceived const& reply_event)
 {
   const scanner_reply::Message msg{ scanner_reply::deserialize(reply_event.data_) };
```

Comment:
This statement can throws a `CRCMismatch` exception. The default exception handler is:
```
    template <class FSM,class Event>
    void exception_caught (Event const&,FSM&,std::exception& )
    {
        BOOST_ASSERT(false);
    }
```

Do we have to be concerned about stopping the scanner?